terror_error.go

```diff
@@ -122,16 +115,50 @@ func (e *Error) MessageTemplate() string {
 	return e.message
 }
 
+// NewError creates a standard error object.
+func NewError(code int, message string) *Error {
+	return &Error{
+		code:    ErrCode(code),
+		message: message,
+	}
+}
+
+// NewErrorWithText creates a standard error object using text error code.
+func NewErrorWithText(code string, message string) *Error {
+	return &Error{
+		codeText: ErrCodeText(code),
+		message:  message,
+	}
+}
+
+// SetWorkaround sets the workaround for standard error.
+func (e *Error) SetWorkaround(workaround string) *Error {
+	e.Workaround = workaround
```

Comment:
Can we allow users to dynamically change the fields?

Reply:
This is a decorator function. Since old error in TiDB do not have workaround field. Using this method can add workaround with miniumn change.

Reply:
Reference to the demo https://play.golang.org/p/rJBVbbkbXLP, we can minimize the change and make the error immutable.